src/test/java/seedu/address/storage/JsonAdaptedTransactionTest.java

```diff
@@ -0,0 +1,178 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.storage.JsonAdaptedTransaction.MISSING_FIELD_MESSAGE_FORMAT;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalTransactions.SHOPPING;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.transaction.Amount;
+import seedu.address.model.transaction.Category;
+import seedu.address.model.transaction.DateTime;
+import seedu.address.model.transaction.Location;
+import seedu.address.model.transaction.Name;
+import seedu.address.model.transaction.Type;
+
+public class JsonAdaptedTransactionTest {
```

Comment:
Does this class test for situations where the data is stored as "-"? It looks like if a category or location stores "-", and a user loads it, it could cause an error and not be parsed

Reply:
Location handles `-` already with the Regex, I think Category isn't handling it right, lemme fix that

Reply:
Addressed in latest push